`gardenctl download tf infra` reports error when current stack is garden+seed+namespace

**Describe the bug**
say we have current target stack:
```
~/git_repo/gardenctl(fix-download-error*) » gardenctl get target                                                           (ccee-m3/default)
target:
- kind: garden
  name: xxx
- kind: seed
  name: xxxx
- kind: namespace
  name: xxxxxxx
```
the command `gardenctl download tf infra` will report following error:

```
~/git_repo/gardenctl(add-test-cases*) » gardenctl download tf infra                                                        (ccee-m3/default)
Kubeconfig under path /Users/i352986/.garden/cache/live/seeds/az-eu2/kubeconfig.yaml contains exec configurations that could contain malicious code. Please only continue if you have verified it to be uncritical
Kubeconfig under path /Users/i352986/.garden/cache/live/seeds/az-eu2/kubeconfig.yaml contains exec configurations that could contain malicious code. Please only continue if you have verified it to be uncritical
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1fe837e]

goroutine 1 [running]:
github.com/gardener/gardenctl/pkg/cmd.downloadTerraformFiles(0x7ffeefbfed76, 0x5, 0xc00041fcf0, 0x114f6a4)
	/Users/i352986/git_repo/gardenctl/pkg/cmd/download.go:95 +0x5ae
github.com/gardener/gardenctl/pkg/cmd.NewDownloadCmd.func1(0xc0003ff340, 0xc0002d9380, 0x2, 0x2)
	/Users/i352986/git_repo/gardenctl/pkg/cmd/download.go:44 +0xed
github.com/spf13/cobra.(*Command).execute(0xc0003ff340, 0xc0002d92e0, 0x2, 0x2, 0xc0003ff340, 0xc0002d92e0)
	/Users/i352986/git_repo/gardenctl/vendor/github.com/spf13/cobra/command.go:844 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x30a3500, 0x1d, 0xc0001a3b40, 0x36)
	/Users/i352986/git_repo/gardenctl/vendor/github.com/spf13/cobra/command.go:945 +0x317
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/i352986/git_repo/gardenctl/vendor/github.com/spf13/cobra/command.go:885
github.com/gardener/gardenctl/pkg/cmd.Execute()
	/Users/i352986/git_repo/gardenctl/pkg/cmd/root.go:112 +0x3aa
main.main()
	/Users/i352986/git_repo/gardenctl/cmd/gardenctl/main.go:20 +0x20
```

**To Reproduce**
Steps to reproduce the behavior:
1. Which target was set 'gardenctl get target'
```
target:
- kind: garden
  name: xxx
- kind: seed
  name: xxxx
- kind: namespace
  name: xxxxxxx
```
2. Which command was entered [e.g. 'gardenctl show vpn-seed']
`gardenctl download tf infra`
3. What was the output of the command 
`[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1fe837e]`
**Expected behavior**
this command should report "no shoots targeted"

**Screenshots**
![image](https://user-images.githubusercontent.com/42594392/91784231-f6207c00-ec34-11ea-997d-1e9dfbdda15e.png)


**Gardenctl Version (please complete the following information):**
 - Version/Commit [c62e0be1]

**Additional context**

similar error also exist in `gardenctl logs vpn-seed`, so fixing them together
```
 gardenctl logs vpn-seed                                                        (ccee-m3/default)
-----------------------Kube-Apiserver
Kubeconfig under path /Users/i352986/.garden/cache/live/seeds/az-eu2/kubeconfig.yaml contains auth provider configurations that could contain malicious code. Please only continue if you have verified it to be uncritical
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x308 pc=0x1ffd0be]

goroutine 1 [running]:
github.com/gardener/gardenctl/pkg/cmd.getSeedNamespaceNameForShoot(0xc0003d47a0, 0x16, 0xc0002e74e0, 0x1049100)
	/Users/i352986/git_repo/gardenctl/pkg/cmd/miscellaneous.go:211 +0x2fe
github.com/gardener/gardenctl/pkg/cmd.logPod(0x230d3c4, 0xe, 0x23055ac, 0x4, 0x2307d1e, 0x8)
	/Users/i352986/git_repo/gardenctl/pkg/cmd/logs.go:196 +0xaa
github.com/gardener/gardenctl/pkg/cmd.logsVpnSeed()
	/Users/i352986/git_repo/gardenctl/pkg/cmd/logs.go:480 +0xbc
github.com/gardener/gardenctl/pkg/cmd.runCommand(0xc0003e22d0, 0x1, 0x1)
	/Users/i352986/git_repo/gardenctl/pkg/cmd/logs.go:148 +0x404
github.com/gardener/gardenctl/pkg/cmd.NewLogsCmd.func2(0xc0003db8c0, 0xc0003e22d0, 0x1, 0x1)
	/Users/i352986/git_repo/gardenctl/pkg/cmd/logs.go:58 +0x3f
github.com/spf13/cobra.(*Command).execute(0xc0003db8c0, 0xc0003e2290, 0x1, 0x1, 0xc0003db8c0, 0xc0003e2290)
	/Users/i352986/git_repo/gardenctl/vendor/github.com/spf13/cobra/command.go:844 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x30a3500, 0x1d, 0xc00012fcc0, 0x36)
	/Users/i352986/git_repo/gardenctl/vendor/github.com/spf13/cobra/command.go:945 +0x317
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/i352986/git_repo/gardenctl/vendor/github.com/spf13/cobra/command.go:885
github.com/gardener/gardenctl/pkg/cmd.Execute()
	/Users/i352986/git_repo/gardenctl/pkg/cmd/root.go:112 +0x3aa
main.main()
	/Users/i352986/git_repo/gardenctl/cmd/gardenctl/main.go:20 +0x20
```

--- a/pkg/cmd/download.go
+++ b/pkg/cmd/download.go
@@ -62,6 +62,10 @@ func downloadTerraformFiles(option string) string {
 	if getRole() == "user" {
 		gardenName := target.Stack()[0].Name
 		projectName := target.Stack()[1].Name
+		if (len(target.Stack()) < 3) || (len(target.Stack()) == 3 && target.Stack()[2].Kind == "namespace") {
+			fmt.Println("No Shoot targeted")
+			os.Exit(2)
+		}
 		shootName := target.Stack()[2].Name
 		pathTerraform := filepath.Join(pathGardenHome, "cache", gardenName, "projects", projectName, shootName)
 		return filepath.Join(pathGardenHome, pathTerraform)
@@ -73,7 +77,7 @@ func downloadTerraformFiles(option string) string {
 	gardenName := target.Stack()[0].Name
 	pathSeedCache := filepath.Join("cache", gardenName, "seeds")
 	pathProjectCache := filepath.Join("cache", gardenName, "projects")
-	if len(target.Stack()) < 3 && (option == "infra" || option == "internal-dns" || option == "external-dns" || option == "ingress" || option == "backup") {
+	if (len(target.Stack()) < 3 && (option == "infra" || option == "internal-dns" || option == "external-dns" || option == "ingress" || option == "backup")) || (len(target.Stack()) == 3 && target.Stack()[2].Kind == "namespace") {
 		fmt.Println("No Shoot targeted")
 		os.Exit(2)
 	} else if len(target.Stack()) < 3 {

--- a/pkg/cmd/logs.go
+++ b/pkg/cmd/logs.go
@@ -189,7 +189,7 @@ func runCommand(args []string) {
 func logPod(toMatch string, toTarget string, container string) {
 	var target Target
 	ReadTarget(pathTarget, &target)
-	if len(target.Target) < 3 {
+	if len(target.Target) < 3 || (len(target.Stack()) == 3 && target.Stack()[2].Kind == "namespace") {
 		fmt.Println("No shoot targeted")
 		os.Exit(2)
 	}

--- a/pkg/cmd/show.go
+++ b/pkg/cmd/show.go
@@ -45,7 +45,7 @@ func NewShowCmd() *cobra.Command {
 			checkError(err)
 			err = yaml.Unmarshal(targetFile, &t)
 			checkError(err)
-			if len(t.Target) < 3 && (args[0] != "operator") && (args[0] != "tf") && (args[0] != "kubernetes-dashboard") && (args[0] != "etcd-operator") && (args[0] != "kibana") {
+			if (len(t.Target) < 3 || (len(t.Target) == 3 && t.Stack()[2].Kind == "namespace")) && (args[0] != "operator") && (args[0] != "tf") && (args[0] != "kubernetes-dashboard") && (args[0] != "etcd-operator") && (args[0] != "kibana") {
 				fmt.Println("No shoot targeted")
 				os.Exit(2)
 			} else if (len(t.Target) < 2 && (args[0] == "tf")) || len(t.Target) < 3 && (args[0] == "tf") && (t.Target[1].Kind != "seed") || (len(t.Target) < 2 && (args[0] == "kibana")) {

